src/zenml/client.py

```diff
@@ -3140,6 +3147,331 @@ def list_run_metadata(
+            updated=updated,
+        )
+        secret_filter_model.set_scope_workspace(self.active_workspace.id)
+        return self.zen_store.list_secrets(
+            secret_filter_model=secret_filter_model
+        )
+
+    def list_secrets_in_scope(
+        self,
+        scope: SecretScope,
+    ) -> Page[SecretResponseModel]:
+        """Fetches the list of secret in a given scope.
+
+        Args:
+            scope: The secrets scope to search for.
+
+        Returns:
+            The list of secrets.
+        """
+        logger.debug(f"Fetching the secrets in scope {scope.value}.")
+
+        return self.list_secrets(
+            scope=scope,
+        )
+
+    def update_secret(
+        self,
+        name_id_or_prefix: str,
+        scope: Optional[SecretScope] = None,
+        new_name: Optional[str] = None,
+        new_scope: Optional[SecretScope] = None,
+        add_or_update_values: Optional[Dict[str, str]] = None,
+        remove_values: Optional[List[str]] = None,
+    ) -> SecretResponseModel:
+        """Updates a secret.
+
+        Args:
+            name_id_or_prefix: The name, id or prefix of the id for the
+                secret to update.
+            scope: The scope of the secret to update.
+            new_name: The new name of the secret.
+            new_scope: The new scope of the secret.
+            add_or_update_values: The values to add or update.
+            remove_values: The values to remove.
+
+        Returns:
+            The updated secret.
+        """
+        secret = self.get_secret(
+            name_id_or_prefix=name_id_or_prefix,
+            scope=scope,
+            # Don't allow partial name matches, but allow partial ID matches
+            allow_partial_name_match=False,
+            allow_partial_id_match=True,
+        )
+
+        secret_update = SecretUpdateModel()
+
+        if new_name:
+            secret_update.name = new_name
+        if new_scope:
+            secret_update.scope = new_scope
+        values: Dict[str, Optional[str]] = {}
+        if add_or_update_values:
+            values.update(add_or_update_values)
+        if remove_values:
+            for key in remove_values:
+                values[key] = None
+        if values:
+            secret_update.values = cast(Dict[str, Optional[SecretStr]], values)
+
+        return Client().zen_store.update_secret(
+            secret_id=secret.id, secret_update=secret_update
+        )
+
+    def delete_secret(
+        self, name_id_or_prefix: str, scope: Optional[SecretScope] = None
+    ) -> None:
+        """Deletes a secret.
+
+        Args:
+            name_id_or_prefix: The name or ID of the secret.
+            scope: The scope of the secret to delete.
+        """
+        secret = self.get_secret(
+            name_id_or_prefix=name_id_or_prefix,
+            scope=scope,
+            # Don't allow partial name matches, but allow partial ID matches
+            allow_partial_name_match=False,
```

Comment:
Is this something that you think makes sense for other delete methods as well, or is it more important in the case of secrets to avoid accidently deleting something?

Reply:
Yes, this is something that I think may be confusing in a lot of other places. Prefix matching is fine when you retrieve information with get/list, but in the case of update/delete operations, I feel like prefix matching on names should be turned off, otherwise you may end up updating/deleting the wrong object. This is how Docker does it, for reference. I have a ticket already filed for this.